### PR TITLE
[MRESOLVER-184] Destroy Redisson semaphores if not used anymore

### PR DESCRIPTION
**Pull request by michael-o**

### CODE REVIEW

File: maven-resolver-named-locks-redisson/src/main/java/org/eclipse/aether/named/redisson/RedissonSemaphoreNamedLockFactory.java [hunk 1]
```diff
@@ -37,12 +41,31 @@
 {
     public static final String NAME = "semaphore-redisson";
 
+    private final ConcurrentMap<String, RSemaphore> semaphores;
+
+    @Inject
+    public RedissonSemaphoreNamedLockFactory()
+    {
+        super();
```
Comment (cstamas) [inline maven-resolver-named-locks-redisson/src/main/java/org/eclipse/aether/named/redisson/RedissonSemaphoreNamedLockFactory.java:49]: ?

**Comment by cstamas** [comment]
IMO -1 for this PR (but +1 for removal of semaphores, but let's do it differently), let me explain:
a) you do same thing as lock factory (map of locks by name), repeating 
b) lets add new method (ie. `destroy()`) on NamedLockSupport that is by default noop, and is overridden by those (redis, hz) that needs to do something when lock destroyed by factory (as ref count goes to 0).

This way, all this PR becomes just a change in `createLock` method, where you override `destroy` method of created NamedLockSupport. WDYT?

**Comment by cstamas** [comment]
Alternate PR https://github.com/apache/maven-resolver/pull/110

**Comment by michael-o** [comment]
Superseded by #111.

**Comment by jira-importer** [comment]
Resolve #1416

**Pull request closed without merging**